tests/unit/hosts/test_views.py

```diff
@@ -328,116 +328,236 @@ def test_create_challenge_host_team_with_no_data(self):
+#                 "pk": self.challenge_host.pk + 1,
+#             },
+#         )
+#         expected = {"error": "ChallengeHost does not exist"}
+#         response = self.client.get(self.url, {})
+#         self.assertEqual(response.data, expected)
+#         self.assertEqual(response.status_code, status.HTTP_406_NOT_ACCEPTABLE)
+
+#     def test_particular_challenge_host_team_for_challenge_host_does_not_exist(
+#         self
+#     ):
+#         self.url = reverse_lazy(
+#             "hosts:get_challenge_host_details",
+#             kwargs={
+#                 "challenge_host_team_pk": self.challenge_host_team.pk + 1,
+#                 "pk": self.challenge_host.pk,
+#             },
+#         )
+#         expected = {"error": "ChallengeHostTeam does not exist"}
+#         response = self.client.get(self.url, {})
+#         self.assertEqual(response.data, expected)
+#         self.assertEqual(response.status_code, status.HTTP_406_NOT_ACCEPTABLE)
+
+
+# class UpdateParticularChallengeHost(BaseAPITestClass):
+#     def setUp(self):
+#         super(UpdateParticularChallengeHost, self).setUp()
+#         self.url = reverse_lazy(
+#             "hosts:get_challenge_host_details",
+#             kwargs={
+#                 "challenge_host_team_pk": self.challenge_host_team.pk,
+#                 "pk": self.challenge_host.pk,
+#             },
+#         )
+#         self.data = {
+#             "status": ChallengeHost.SELF,
+#             "permissions": ChallengeHost.WRITE,
+#         }
+#         self.partial_update_data = {"status": ChallengeHost.DENIED}
+
+#     def test_particular_challenge_host_partial_update(self):
+#         expected = {
+#             "id": self.challenge_host.pk,
+#             "user": self.challenge_host.user.username,
+#             "team_name": self.challenge_host.team_name.pk,
+#             "status": ChallengeHost.DENIED,
+#             "permissions": self.challenge_host.permissions,
+#         }
+#         response = self.client.patch(self.url, self.partial_update_data)
+#         self.assertEqual(response.data, expected)
+#         self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+#     def test_particular_challenge_host_update(self):
+#         expected = {
+#             "id": self.challenge_host.pk,
+#             "user": self.challenge_host.user.username,
+#             "team_name": self.challenge_host.team_name.pk,
+#             "status": ChallengeHost.SELF,
+#             "permissions": ChallengeHost.WRITE,
+#         }
+#         response = self.client.put(self.url, self.data)
+#         self.assertEqual(response.data, expected)
+#         self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+#     def test_particular_challenge_host_update_with_no_data(self):
+#         self.data = {"status": ""}
+#         response = self.client.put(self.url, self.data)
+#         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
```

Comment:
@Kajol-Kumari Please don't remove existing tests. The updates should work with original tests as well.
>
>  Commenting due to absence of url to these tests along with these two test cases are pointing to the same url name i.e get_challenge_host_details.

So the URL you renamed above is for both functionalities. It will go through the lines of code you wrote for deleting when it is used with `DELETE` method and will execute differently when the method is `GET`.
Your additions should be compatible with both. :)